core/src/main/java/nl/nn/adapterframework/jdbc/migration/LiquibaseMigrator.java

```diff
@@ -0,0 +1,194 @@
+import liquibase.Liquibase;
+import liquibase.Scope;
+import liquibase.change.CheckSum;
+import liquibase.changelog.ChangeLogHistoryServiceFactory;
+import liquibase.changelog.ChangeSet;
+import liquibase.changelog.DatabaseChangeLog;
+import liquibase.changelog.RanChangeSet;
+import liquibase.database.Database;
+import liquibase.database.DatabaseConnection;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.exception.LiquibaseException;
+import liquibase.exception.LockException;
+import liquibase.exception.ValidationFailedException;
+import liquibase.executor.ExecutorService;
+import liquibase.lockservice.LockService;
+import liquibase.lockservice.LockServiceFactory;
+import liquibase.resource.ResourceAccessor;
+import nl.nn.adapterframework.configuration.ConfigurationWarnings;
+import nl.nn.adapterframework.jdbc.JdbcException;
+import nl.nn.adapterframework.util.AppConstants;
+import nl.nn.adapterframework.util.LogUtil;
+
+/**
+ * LiquiBase implementation for IAF
+ * 
+ * @author	Niels Meijer
+ * @since	7.0-B4
+ *
+ */
+public class LiquibaseMigrator extends DatabaseMigratorBase {
+
+	protected Logger migrationLog = LogUtil.getLogger("liquibase.migrationLog");
+	private Contexts contexts;
+	private LabelExpression labelExpression = new LabelExpression();
+
+	private String getChangeLogFile() throws IOException {
+		AppConstants appConstants = AppConstants.getInstance(getApplicationContext().getClassLoader());
+		String changeLogFile = appConstants.getString("liquibase.changeLogFile", "DatabaseChangelog.xml");
+
+		if(getResource(changeLogFile) == null) {
+			String msg = "unable to find database changelog file [" + changeLogFile + "]";
+			msg += " classLoader [" + getConfigurationClassLoader() + "]";
+			throw new IOException(msg);
+		}
+
+		return changeLogFile;
+	}
+
+	private Liquibase createMigrator(InputStream file) throws IOException, SQLException, LiquibaseException {
+		String changeLogFile = getChangeLogFile();
+
+		ResourceAccessor resourceAccessor;
+		if(file == null) {
+			resourceAccessor = new LiquibaseResourceAccessor(getConfigurationClassLoader());
+		} else {
+			resourceAccessor = new StreamResourceAccessor(file);
+		}
+		DatabaseConnection connection = getDatabaseConnection();
+
+		return new Liquibase(changeLogFile, resourceAccessor, connection);
+	}
+
+	private DatabaseConnection getDatabaseConnection() throws SQLException {
+		return new JdbcConnection(lookupMigratorDatasource().getConnection());
+	}
+
+	@Override
+	public boolean validate() {
+		try {
+			doValidate();
+			return true;
+		}
+		catch (ValidationFailedException e) {
+			ConfigurationWarnings.add(this, log, "liquibase validation failed: "+e.getMessage(), e);
+		}
+		catch (LiquibaseException e) {
+			ConfigurationWarnings.add(this, log, "liquibase failed to initialize", e);
+		}
+		catch (IOException e) {
+			log.debug(e.getMessage(), e); //this can only happen when jdbc.migrator.active=true but no migrator file is present.
```

Comment:
Misschien handiger om in het geval er geen migrator file is (Ik neem aan dat daarmee een changelog.xml bedoeld wordt) niet een exceptie te gooien, om de IOExceptie als die er wel is als warning te kunnen behandelen.